src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,165 @@
+# pylint: disable=too-many-lines
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import sre_parse
+from typing import List, Optional
+import shutil
+import subprocess
+import requests
+import os
+import platform
+from pathlib import Path
+from knack.prompting import prompt_y_n
+
+
+def aks_draft_app_init(deployment_path: str,
+                       app_name: str,
+                       language: str,
+                       create_config: str,
+                       dockerfile_only: str,
+                       deployment_only: str) -> None:
+    file_path = _binary_pre_check()
+    if not file_path:
+        raise ValueError('Binary check was NOT executed successfully')
+
+    arguments = _build_arguments(app_name, language, create_config, dockerfile_only, deployment_only)
+    run_successful = _run(file_path, deployment_path, arguments)
+    if run_successful:
+        _cmd_finish()
+    else:
+        raise ValueError('\'az aks app init\' was NOT executed successfully')
+
+
+# If setup is valid this method returns the correct binary path to execute
+def _binary_pre_check() -> str:
+    print('The DraftV2 setup is in progress...')
+    draftv2_binary_path = _get_existing_path()
+
+    if draftv2_binary_path:  # use found binary
+        return draftv2_binary_path
+    else:  # prompt user to download binary
+        return _download_binary()
+
+
+def _get_filename() -> Optional[str]:
+    operating_system = platform.system().lower()
+    architecture = platform.machine().lower()
+
+    if architecture == 'x86_64':
+        architecture = 'amd64'
+    if architecture not in ['arm64', 'amd64']:
+        print('Cannot find a suitable download for the current system architecture. Draft only supports AMD64 and ARM64.')
+        return None
+
+    return f'draftv2-{operating_system}-{architecture}'
+
+
+# Returns path to existing draftv2 binary and None otherwise
+def _get_existing_path() -> Optional[str]:
+    print('Checking if DraftV2 binary exists locally...')
+
+    filename = _get_filename()
+    if not filename:
+        return None
+
+    paths = _get_potential_paths()
+    if not paths:
+        print('List of potential DraftV2 paths is empty')
+        return None
+
+    for path in paths:
+        binary_file_path = path + '/' + filename
+        if os.path.exists(binary_file_path):
+            print('Existing binary found at: ' + binary_file_path)
+            return binary_file_path
+    return None
+
+
+# Returns a list of potential draftV2 binary paths
+def _get_potential_paths() -> List[str]:
+    result = [str(Path.home()) + '/' + '.aksapp']
+    paths = os.environ['PATH'].split(':')
```

Comment:
I think you should just return a concatenation of `paths` and `result`. This will allow you to detect the draftv2 binary anywhere along the customer's PATH. Otherwise we're downloading the draftv2 binary for a second time, when it's already executable, since it's in one of the PATH directories